src/java/grl/src/grl/driver/GRL_Driver_ChangeModes.java

```diff
@@ -0,0 +1,1144 @@
+import grl.flatbuffer.ArmState;
+import grl.flatbuffer.CartesianImpedenceControlMode;
+import grl.flatbuffer.KUKAiiwaInterface;
+import grl.flatbuffer.KUKAiiwaMonitorState;
+import grl.flatbuffer.KUKAiiwaState;
+import grl.flatbuffer.KUKAiiwaStates;
+import grl.flatbuffer.MoveArmJointServo;
+import grl.flatbuffer.MoveArmTrajectory;
+import grl.flatbuffer.Wrench;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+import com.google.flatbuffers.FlatBufferBuilder;
+import com.google.flatbuffers.Table;
+import com.kuka.common.ThreadUtil;
+import com.kuka.connectivity.fastRobotInterface.FRIConfiguration;
+import com.kuka.connectivity.fastRobotInterface.FRIJointOverlay;
+import com.kuka.connectivity.fastRobotInterface.FRISession;
+import com.kuka.connectivity.motionModel.smartServo.ISmartServoRuntime;
+import com.kuka.connectivity.motionModel.smartServo.ServoMotion;
+import com.kuka.connectivity.motionModel.smartServo.ServoMotionJP;
+import com.kuka.connectivity.motionModel.smartServo.SmartServo;
+//import com.kuka.generated.ioAccess.FlexFellowIOGroup;
+//import com.kuka.generated.ioAccess.MediaFlangeIOGroup;
+import com.kuka.roboticsAPI.applicationModel.RoboticsAPIApplication;
+import com.kuka.roboticsAPI.controllerModel.Controller;
+import com.kuka.roboticsAPI.controllerModel.recovery.IRecovery;
+import com.kuka.roboticsAPI.deviceModel.JointLimits;
+import com.kuka.roboticsAPI.deviceModel.JointPosition;
+import com.kuka.roboticsAPI.deviceModel.LBR;
+import com.kuka.roboticsAPI.executionModel.CommandInvalidException;
+import com.kuka.roboticsAPI.executionModel.ExecutionState;
+import com.kuka.roboticsAPI.executionModel.IExecutionContainer;
+import com.kuka.roboticsAPI.geometricModel.CartDOF;
+import com.kuka.roboticsAPI.geometricModel.LoadData;
+import com.kuka.roboticsAPI.geometricModel.PhysicalObject;
+import com.kuka.roboticsAPI.geometricModel.Tool;
+import com.kuka.roboticsAPI.motionModel.HandGuidingMotion;
+import com.kuka.roboticsAPI.motionModel.IMotion;
+import com.kuka.roboticsAPI.motionModel.IMotionContainer;
+import com.kuka.roboticsAPI.motionModel.IMotionContainerListener;
+import com.kuka.roboticsAPI.motionModel.MotionBatch;
+import com.kuka.roboticsAPI.motionModel.controlModeModel.AbstractMotionControlMode;
+import com.kuka.roboticsAPI.motionModel.controlModeModel.CartesianImpedanceControlMode;
+import com.kuka.roboticsAPI.motionModel.controlModeModel.CartesianSineImpedanceControlMode;
+import com.kuka.roboticsAPI.motionModel.controlModeModel.HandGuidingControlMode;
+import com.kuka.roboticsAPI.motionModel.controlModeModel.JointImpedanceControlMode;
+import com.kuka.roboticsAPI.motionModel.controlModeModel.PositionControlMode;
+import static com.kuka.roboticsAPI.motionModel.MMCMotions.*;
+
+/**
+ * Creates a FRI Session.
+ */
+public class GRL_Driver_ChangeModes extends RoboticsAPIApplication

```

Comment:
Can a more detailed explanation of this class be added here? What does it do? Why is it needed? How do I interact with it? Where is it used and why?